tests/drivers/can/api/src/classic.c

```diff
@@ -576,6 +635,100 @@ ZTEST_USER(can_classic, test_set_timing_max)
 	zassert_equal(err, 0, "failed to start CAN controller (err %d)", err);
 }
 
+/**
+ * @brief Test that timing values for the data phase that are above the maximum can't be set.
+ */
+ZTEST_USER(can_classic, test_set_timing_above_max)
+{
+	int err;
+
+	const struct can_timing * max_timing = can_get_timing_max(can_dev);
+	struct can_timing test_timing;
+
+	err = can_stop(can_dev);
+	zassert_equal(err, 0, "failed to stop CAN controller (err %d)", err);
+
+	test_timing = *max_timing;
+	test_timing.sjw += 1;
+	err = can_set_timing(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid sjw was accepted (err %d)", err);
+
+	test_timing = *max_timing;
+	test_timing.prop_seg += 1;
+	err = can_set_timing(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid prop_seg was accepted (err %d)", err);
+
+	test_timing = *max_timing;
+	test_timing.phase_seg1 += 1;
+	err = can_set_timing(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid phase_seg1 was accepted (err %d)", err);
+
+	test_timing = *max_timing;
+	test_timing.phase_seg2 += 1;
+	err = can_set_timing(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid phase_seg2 was accepted (err %d)", err);
+
+	test_timing = *max_timing;
+	test_timing.prescaler += 1;
+	err = can_set_timing(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid prescaler was accepted (err %d)", err);
+
+	err = can_start(can_dev);
+	zassert_equal(err, 0, "failed to start CAN controller (err %d)", err);
+}
```

Comment:
Same as above.

Reply:
These tests cover
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/can/can_common.c#L392
and
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/can/can_common.c#L368
Current implementation of CAN driver provides ways to set min/max values for sample point.
Since, driver uses these values, I think that it shall be tested.